### PR TITLE
chore/SOF-1550

### PR DESCRIPTION
**Pull request by spectre007**

### CODE REVIEW

File: src/lattice/reciprocal/lattice_reciprocal.js [hunk 1]
```diff
@@ -69,4 +87,28 @@ export class ReciprocalLattice extends Lattice {
         });
         return kpointPath;
     }
+
+    /**
+     * Calculate grid dimension based on reciprocal lattice vectors.
+     * @param nPoints - Total number of points
+     * @param index - Index of reciprocal vector
```
Comment (timurbazhirov) [inline src/lattice/reciprocal/lattice_reciprocal.js:94]: Docstrings should have types:

```js
    /**
     * @summary Save unit's results and monitors.
     * @param job {Job} Job object.
     * @param flowchartId {String} Unit flowchartId.
     * @param payload {Object} Unit result.
     * @param header {Object} Unit result header.
     * @private
     */
```

**Review by timurbazhirov — APPROVED** [review]

File: src/lattice/reciprocal/lattice_reciprocal.js [hunk 2]
```diff
@@ -69,4 +87,28 @@ export class ReciprocalLattice extends Lattice {
...
+        const N = Math.cbrt((nPoints * norms[index] ** 2) / (norms[j] * norms[k]));
+        return Math.max(1, Math.ceil(N));
+    }
+
+    /**
+     * Calculate grid dimensions from total number of k-points.
+     * @param nKpoints - Total number of k-points.
+     * @return {number[]} - Grid dimensions
+     */
+    getDimensionsFromPoints(nKpoints) {
```
Comment (timurbazhirov) [inline src/lattice/reciprocal/lattice_reciprocal.js:110]: getDimensionsFromPointsCount maybe?

File: src/lattice/reciprocal/lattice_reciprocal.js [hunk 3]
```diff
@@ -69,4 +87,28 @@ export class ReciprocalLattice extends Lattice {
...
+     * Calculate grid dimension based on reciprocal lattice vectors.
+     * @param nPoints - Total number of points
+     * @param index - Index of reciprocal vector
+     * @return {number} - Grid dimension in direction of reciprocal vector
+     * @todo This could be moved to a separate KGrid class.
+     */
+    calculateDimension(nPoints, index) {
+        const norms = this.reciprocalVectorNorms;
+        const [j, k] = [0, 1, 2].filter((i) => i !== index); // get indices of other two dimensions
+        const N = Math.cbrt((nPoints * norms[index] ** 2) / (norms[j] * norms[k]));
```
Comment (timurbazhirov) [inline src/lattice/reciprocal/lattice_reciprocal.js:101]: Let's just use `math` maybe as we importing it everywhere? It wraps `Math` also, I think

Comment (spectre007) [reply src/lattice/reciprocal/lattice_reciprocal.js:101]: `math` wraps [math.js](https://mathjs.org/docs/reference/functions.html) v3.20.2, but not `Math`. Ok, I'll change it to use `math` only.

**Pull request merged**